work14/index.js

```diff
@@ -0,0 +1,120 @@
+"use strict";
+const modalBtn = document.getElementById("js_modalButton");
+const modal = document.getElementById("js_modal");
+const requestBtn = document.getElementById("js_requestButton");
+
+const url = "https://mocki.io/v1/d6da0b8a-3546-419e-aaec-05f3a247c6d0";
+// const url = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// const url = "https://myjson.dit.upm.es/api/bins/eu2f";
+
+// 下記は、myjson繋がらない時の固定値
+// const url = {
+//   "data": [
+//     {
+//       "a": "bookmark",
+//       "img": "img/1.png",
+//       "alt": "画像１",
+//       "text": "ブックマーク"
+//     },
+//     {
+//       "a": "message",
+//       "img": "img/2.png",
+//       "alt": "画像２",
+//       "text": "メッセージ"
+//     }
+//   ]
+// }
+
+async function init() {
+  renderLoading();
+  const data = await getListData();
+  renderListElement(data);
+}
+
+async function getData() {
+  try {
+    const response = await fetch(url);
+    if (response.ok) {
+      const json = await response.json();
+      return json;
+    } else {
+      throw new Error(`Server request failed:${response.statusText}`);
+    }
+    // 下記は、固定値をそのままpromiseの返り値とする
+    // const json = url
+    // return json
+  } catch (e) {
+    console.error(e)
+    throw new Error(e);
+  }
+}
+
+async function getListData() {
+  let listData;
+  const wrap = document.getElementById("js_wrap");
+
+  try {
+    listData = await getData();
+  } catch (e) {
+    wrap.textContent = `エラー内容:${e.message}`;
+  } finally {
+    removeLoading();
+  }
+  if (listData.data.length === 0) {
+    wrap.textContent = "data is empty";
+    return;
+  }
+  return listData;
+}
```

Comment:
IMO
I thought changing `getListData()` to `init()` would make the code a little easier to read.
This allows us to remove the originally defined `init()`.
```suggestion
async function init() {
  renderLoading();
  let listData;
  const wrap = document.getElementById("js_wrap");

  try {
    listData = await getData();
  } catch (e) {
    wrap.textContent = `${e.message}`;
  } finally {
    removeLoading();
  }
  if (listData.data.length === 0) {
    wrap.textContent = "data is empty";
  }
  renderListElement(listData);
}
```